js/genkit/src/genkit.ts

```diff
@@ -271,29 +271,33 @@ export class Genkit {
    *
    * @todo TODO: Show an example of a name and variant.
    */
-  async prompt<
+  prompt<
     I extends z.ZodTypeAny = z.ZodTypeAny,
     O extends z.ZodTypeAny = z.ZodTypeAny,
     CustomOptions extends z.ZodTypeAny = z.ZodTypeAny,
   >(
     name: string,
     options?: { variant?: string }
-  ): Promise<ExecutablePrompt<z.infer<I>, O, CustomOptions>> {
-    // check the registry first as not all prompt types can be
-    // loaded by dotprompt (e.g. functional)
-    let action = (await this.registry.lookupAction(
-      `/prompt/${name}`
-    )) as PromptAction<I>;
-    // nothing in registry - check for dotprompt file.
-    if (!action) {
-      action = (await prompt(this.registry, name, options))
-        .promptAction as PromptAction<I>;
-    }
+  ): ExecutablePrompt<z.infer<I>, O, CustomOptions> {
+    const actionPromise = (async () => {
+      // check the registry first as not all prompt types can be
+      // loaded by dotprompt (e.g. functional)
+      let action = (await this.registry.lookupAction(
+        `/prompt/${name}`
+      )) as PromptAction<I>;
+      // nothing in registry - check for dotprompt file.
+      if (!action) {
+        action = (await prompt(this.registry, name, options))
+          .promptAction as PromptAction<I>;
+      }
```

Comment:
You might want to look at `promptRef` and its implementation...I believe it also accounted for reloading prompts when the files changed but don't remember for sure.

Reply:
I think it's conceptually similar...